libflux/src/semantic/tests.rs

```diff
@@ -204,6 +204,636 @@ fn instantiation_1() {
     }
 }
 #[test]
+fn instantation_literals() {
+    test_infer! {
+        src: r#"
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = 12h
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/
+        "#,
+        exp: &[
+            ("a", "forall [] string"),
+            ("b", "forall [] int"),
+            ("c", "forall [] float"),
+            ("d", "forall [] duration"),
+            ("e", "forall [] time"),
+            ("f", "forall [] regexp"),
+        ],
+    }
+}
+#[test]
+fn instantiate_string_expr() {
+    let src = r#"message = "Hello, ${name}!""#;
+    test_infer! {
+        env: &[
+            ("name", "forall [t0] t0"),
+        ],
+        src: src,
+        exp: &[
+            ("message", "forall [] string"),
+        ],
+    }
+    test_infer_err! {
+        env: &[
+            ("name", "forall [] int"),
+        ],
+        src: src,
+    }
+    test_infer! {
+        env: &[
+            ("name", "forall [] string"),
+        ],
+        src: src,
+        exp: &[
+            ("message", "forall [] string"),
+        ],
+    }
+}
+#[test]
+fn instantiate_array_expr() {
+    test_infer! {
+        src: r#"
+            a = [1, 2, 3]
+        "#,
+        exp: &[
+            ("a", "forall [] [int]"),
+        ],
+    }
+    test_infer! {
+        src: r#"
+            a = ["a", "b", "c"]
+        "#,
+        exp: &[
+            ("a", "forall [] [string]"),
+        ],
+    }
+    test_infer! {
+        env: &[
+            ("a", "forall [t0] t0"),
+            ("b", "forall [t1] t1"),
```

Comment:
As mentioned above, these aren't valid Flux types so I would prefer not to use them in test cases. For this particular test case perhaps you should test that the empty array type is `forall [] [t0]`?

Again, I think the way you want to write this test is like:
```Rust
test_infer! {
    before: &[
        ("a", "forall [] t0"),
        ("b", "forall [] t1"),
    ],
    src: r#"
        c = [a, b]
    "#,
    after: &[
        ("a", "forall [] t2"),
        ("b", "forall [] t2"),
        ("c", "forall [] [t2]"),
    ],
}
```
This is a deficiency of the testing framework and I will address it.

Reply:
This comment is not relevant anymore because of [this comment](https://github.com/influxdata/flux/pull/2102#issuecomment-551848574)